src/ng/compile.js

```diff
@@ -2026,6 +2085,7 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
           }
 
           // use class as directive
+          if (!cssClassDirectivesEnabled) break;
```

Comment:
I think it is safer to cache the value of the configuration when instantiating the service. Currently it is possible to change this value at runtime, which makes the behavior less predictable.
(Unless we have a complelling usecase for supporting this.)